### ISSUE TITLE
LibPack 3.0.0 does not support building in DEBUG mode

### ISSUE THREAD

**Issue by chennes** [FreeCAD/FreeCAD-LibPack#22]
Right now building the LibPack with the `--mode=debug` fails due to a set of issues. First and foremost, the pip install command needs to be modified to pull or compile debug versions of those packages. This is problematic for `numpy`, which requires a Fortran compiler. This Issue is intended as a catch-all to collect problems related to getting the debug-mode LibPack working.

**Comment by oursland** [comment]
We recently made a change to the `Conda Windows Debug` builds to use the [`RelWithDebInfo`](https://github.com/FreeCAD/FreeCAD/blob/main/CMakePresets.json#L280) configuration.  Would it be preferred to make a similar change to the `Debug` configuration when building on Windows?

**Comment by chennes** [comment]
> Would it be preferred to make a similar change to the `Debug` configuration when building on Windows?

To which build? Of FreeCAD?

**Comment by oursland** [comment]
Yes, FreeCAD.  I see now that this issue is specific to LibPack.

**Comment by chennes** [comment]
Right -- for 95% of debugging, RelWithDebInfo is really what you want to be using on Windows, but for that last 5%, it would be nice to have a full debugging build of the whole stack so that FreeCAD can also be compiled with optimization off. I'd certainly welcome help getting the LibPack compiling in Debug, there are some "gotchas" in the process (like OCCT's insistence on making up its own naming convention...).

**Comment by chennes** [comment]
Noting for the record that this remains true as of 3.4.0 -- although compiling the C++ libraries in debug mode is reasonably straightforward, the Python wheels with C++, Fortran, or Rust backends are much more tedious to get working in debug mode.
